lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,323 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import random
+import string
+from pathlib import PurePath, PureWindowsPath
+from typing import Any, List, Optional, Type, cast
+
+from lisa import feature, schema, search_space
+from lisa.environment import Environment
+from lisa.node import RemoteNode
+from lisa.platform_ import Platform
+from lisa.tools import Cp, HyperV, Ls, Mkdir, PowerShell, Unzip
+from lisa.util.logger import Logger, get_logger
+
+from .. import HYPERV
+from .context import get_node_context
+from .schema import HypervNodeSchema, HypervPlatformSchema
+from .serial_console import SerialConsole, SerialConsoleLogger
+
+
+class _HostCapabilities:
+    def __init__(self) -> None:
+        self.core_count = 0
+        self.free_memory_mib = 0
+
+
+class HypervPlatform(Platform):
+    def __init__(
+        self,
+        runbook: schema.Platform,
+    ) -> None:
+        super().__init__(runbook=runbook)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return HYPERV
+
+    @classmethod
+    def supported_features(cls) -> List[Type[feature.Feature]]:
+        return [SerialConsole]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._hyperv_runbook = self._get_hyperv_runbook()
+        self.server_node = self._initialize_server_node()
+        self._host_capabilities = self._init_host_capabilities(self._log)
+        self._common_vhd_path: Optional[PureWindowsPath] = None
+
+    def _get_hyperv_runbook(self) -> HypervPlatformSchema:
+        hyperv_runbook = self.runbook.get_extended_runbook(HypervPlatformSchema)
+        assert hyperv_runbook, "platform runbook cannot be empty"
+        return cast(HypervPlatformSchema, hyperv_runbook)
+
+    def _initialize_server_node(self) -> RemoteNode:
+        assert self._hyperv_runbook, "hyperv runbook cannot be empty"
+
+        if len(self._hyperv_runbook.servers) > 1:
+            self._log.info(
+                "Multiple servers are currently not supported. "
+                "Only the first server will be used."
+            )
+
+        server = self._hyperv_runbook.servers[0]
+        server_node = RemoteNode(
+            runbook=schema.Node(name="hyperv"),
+            index=-1,
+            logger_name="hyperv",
+            parent_logger=get_logger("hyperv"),
+        )
+
+        server_node.set_connection_info(
+            address=server.address, username=server.username, password=server.password
+        )
+
+        return server_node
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        return self._configure_node_capabilities(environment, log)
+
+    def _configure_node_capabilities(
```

Comment:
This method is not shared and meaningful. The name is ambiguous to describe the purpose or executed logic. So, the content should be in `_prepare_environment`.

Reply:
done.